### DISCUSSION TITLE
Question: does Pyodide's numpy use multiple threads?

### DISCUSSION THREAD

**Discussion by tongzhou80** [pyodide/pyodide#5213] [General]
Hi,

I know that numpy is supported by Pyodide, but numpy uses a lot of native code and even MKL libraries for its array operators like matmul. In Pyodide's version, how does it handle the native code/library used by Numpy and how does it handle the multi-threading use of numpy operators? Thanks!

**Comment by ryanking13** [comment]
For now, we don't use any SIMD operations in numpy, no multi-threading as well.

SIMD operations are theoretically possible (some limited number of operations are supported: https://emscripten.org/docs/porting/simd.html), but I believe people from OpenBLAS, MKL, etc. should write a kernel for it.

multi threading is not supported because of the browser limitation
